src/router.ts

```diff
@@ -105,74 +130,52 @@ const Router = {
-    app.use(API_PREFIX + Constants.UPDATES_PATH, nocache, auth,
-            require('./controllers/updates_controller'));
-    app.use(API_PREFIX + Constants.ADDONS_PATH, nocache, auth,
-            require('./controllers/addons_controller'));
-    app.use(API_PREFIX + Constants.RULES_PATH, nocache, auth,
-            require('./rules-engine/index').default);
-    app.use(API_PREFIX + Constants.INTERNAL_LOGS_PATH, nocache, auth,
-            require('./controllers/internal_logs_controller'));
-    app.use(API_PREFIX + Constants.PUSH_PATH, nocache, auth,
-            require('./controllers/push_controller'));
-    app.use(API_PREFIX + Constants.LOGS_PATH, nocache, auth,
-            require('./controllers/logs_controller'));
-    app.use(API_PREFIX + Constants.NOTIFIERS_PATH, nocache, auth,
-            require('./controllers/notifiers_controller'));
-
-    app.use(API_PREFIX + Constants.OAUTH_PATH, nocache,
-            require('./controllers/oauth_controller'));
-    app.use(API_PREFIX + Constants.OAUTHCLIENTS_PATH, nocache, auth,
-            require('./controllers/oauthclients_controller'));
-  },
-
-  addProxyServer(thingId, server) {
-    this.proxyController.addProxyServer(thingId, server);
-  },
-
-  removeProxyServer(thingId) {
-    this.proxyController.removeProxyServer(thingId);
-  },
-};
-
-module.exports = Router;
+    app.use(API_PREFIX + Constants.THINGS_PATH, nocache, auth, ThingsController());
+    app.use(API_PREFIX + Constants.NEW_THINGS_PATH, nocache, auth, NewThingsController());
+    app.use(API_PREFIX + Constants.ADAPTERS_PATH, nocache, auth, AdaptersController());
+    app.use(API_PREFIX + Constants.ACTIONS_PATH, nocache, auth, ActionsController());
+    app.use(API_PREFIX + Constants.EVENTS_PATH, nocache, auth, EventsController());
+    app.use(API_PREFIX + Constants.LOG_OUT_PATH, nocache, auth, LogOutController());
+    app.use(API_PREFIX + Constants.UPLOADS_PATH, nocache, auth, UploadsController());
+    app.use(API_PREFIX + Constants.UPDATES_PATH, nocache, auth, UpdatesController());
+    app.use(API_PREFIX + Constants.ADDONS_PATH, nocache, auth, AddonsController());
+    app.use(API_PREFIX + Constants.RULES_PATH, nocache, auth, RuleEngine);
+    app.use(API_PREFIX + Constants.INTERNAL_LOGS_PATH, nocache, auth, InternalLogsController());
+    app.use(API_PREFIX + Constants.PUSH_PATH, nocache, auth, PushController());
+    app.use(API_PREFIX + Constants.LOGS_PATH, nocache, auth, LogsController());
+    app.use(API_PREFIX + Constants.NOTIFIERS_PATH, nocache, auth, NotifiersController());
+
+    app.use(API_PREFIX + Constants.OAUTH_PATH, nocache, OAuthController());
+    app.use(API_PREFIX + Constants.OAUTHCLIENTS_PATH, nocache, auth, OAuthClientsController());
+  }
+
+  addProxyServer(thingId: string, server: string) {
+    this.proxyStore?.addProxyServer(thingId, server);
+  }
+
+  removeProxyServer(thingId: string) {
+    this.proxyStore?.removeProxyServer(thingId);
+  }
+}
+
+export = new Router();
```

Comment:
I've been trying to use `export default`, rather than `export =`. You'd just need to change the `require()` in `thing.js` and `things.js`.

Reply:
Still need to update the Thing and Things models.

Reply:
`app.js` was missing